MedtrumKit/PumpManager/MedtrumPumpManager.swift

```diff
@@ -4,8 +4,177 @@ import os.log
 
 public class MedtrumPumpManager: DeviceManager {
     public static let pluginIdentifier = "Medtrum"
-
     public let localizedTitle = LocalizedString("Medtrum", comment: "Generic title of the Medtrum pump manager")
+     public let managerIdentifier: String = "MedtrumKit"
+    public var rawState: RawStateValue
+    private let log = MedtrumLogger(category: "MedtrumPumpManager")
+    public let pumpDelegate = WeakSynchronizedDelegate<PumpManagerDelegate>()
+    
+    public required init?(rawState: RawStateValue) {
+        return nil
+    }
+    
+    public var isOnboarded: Bool {
+        false
+    }
+    
+    public static var onboardingMaximumBasalScheduleEntryCount: Int {
+        48
+    }
+    
+    public static var onboardingSupportedBasalRates: [Double] {
+        // 0.05 units for rates between 0.00-25U/hr
+        // 0 U/hr is a supported scheduled basal rate
+        (1 ... 500).map { Double($0) / 20 }
+    }
+
+    public static var onboardingSupportedBolusVolumes: [Double] {
+        // 0.05 units for rates between 0.05-30U
+        // 0 is not a supported bolus volume
+        (1 ... 600).map { Double($0) / 20 }
+    }
+    
+    public static var onboardingSupportedMaximumBolusVolumes: [Double] {
+        MedtrumPumpManager.onboardingSupportedBolusVolumes
+    }
+    
+    public var delegateQueue: DispatchQueue! {
+        get {
+            pumpDelegate.queue
+        }
+        set {
+            pumpDelegate.queue = newValue
+        }
+    }
+    
+    public var supportedBasalRates: [Double] {
+        MedtrumPumpManager.onboardingSupportedBasalRates
+    }
+    
+    public var supportedBolusVolumes: [Double] {
+        MedtrumPumpManager.onboardingSupportedBolusVolumes
+    }
+    
+    public var supportedMaximumBolusVolumes: [Double] {
+        MedtrumPumpManager.onboardingSupportedBolusVolumes
+    }
+    
+    public var maximumBasalScheduleEntryCount: Int {
+        MedtrumPumpManager.onboardingMaximumBasalScheduleEntryCount
+    }
+    
+    public var minimumBasalScheduleEntryDuration: TimeInterval {
+        // One per hour
+        TimeInterval(60 * 60)
```

Comment:
Pump supports segments of 30minutes in the basal profile

Reply:
My bad. Fixed!